src/target/riscv/riscv-013.c

```diff
@@ -4394,182 +4394,299 @@ static int write_memory_bus_v1(struct target *target, target_addr_t address,
 	return ERROR_OK;
 }
 
-static int write_memory_progbuf(struct target *target, target_addr_t address,
-		uint32_t size, uint32_t count, const uint8_t *buffer)
+static uint64_t get_from_buffer_and_log_write(target_addr_t address,
```

Comment:
I would like to leave this name as-is.
This function returns a value from the buffer, so naming it `log_...` would be misleading IMHO.

Adding the `is_read` argument does not seem practical to me, since when memory is read, the buffer is written. This function, on the other hand, reads the buffer.

Reply:
(C) Do we even need this function? If we split the functionality (which seems reasonable), then a user needs to call 2 functions, but this whole thing is already just 2 function calls.

Reply:
Addressed.